pyface/ui/qt4/tasks/editor_area_pane.py

```diff
@@ -82,6 +106,11 @@ def destroy(self):
         for editor in self.editors:
             self.remove_editor(editor)
 
+        if self.control is not None:
```

Comment:
Same here, `self.control.removeEventFilter` is already accessing `control`'s attribute. On the other hand, for the `_connections_to_remove` pattern, I would think `if self.control is not None` is the sort of defensive programming that could backfire...e.g. if the block is not run because of an error that should not have occurred, the `_connections_to_remove` could be holding references that prevent the objects from being garbage collected, leading to even more obscure errors.